api/filters/replacement/replacement.go

```diff
@@ -178,3 +178,27 @@ func makeResId(n *yaml.RNode) *resid.ResId {
 		Namespace: n.GetNamespace(),
 	}
 }
+
+func getFieldPath(fieldPath string) []string {
```

Comment:
Does this belong in kyaml? The features we're supporting here don't really belong to the ReplacementTranformer so much as yaml.PathGetter. Maybe what we want is a variant on the `LookupCreate` PathGetter initializer that takes a raw structured string? Or just a parser helper function to use with the existing LookupCreate.

Reply:
You're right that this logic makes more sense in kyaml. I'll move it.

Reply:
Will move the logic to kyaml in a followup PR